File: vaultx-backend/src/main/java/com/vaultx/service/PublicShareService.java
package com.vaultx.service;

import com.vaultx.dto.share.PublicShareMetadataDto;

import java.io.InputStream;

public interface PublicShareService {
    PublicShareMetadataDto getPublicMetadata(String token, String ipAddress, String userAgent);
    boolean verifyPassword(String token, String password);
    InputStream downloadShare(String token, String password, String ipAddress, String userAgent) throws Exception;
    
    java.util.List<com.vaultx.dto.share.PublicDocumentDto> getShareDocuments(String token, String password, String ipAddress, String userAgent);
    InputStream downloadSingleDocument(String token, String docId, String password, String ipAddress, String userAgent) throws Exception;
}

File: vaultx-frontend/src/services/publicShareService.js
import api from './api';

const publicShareService = {
  getMetadata: (token) => api.get(`/public/shares/${token}`),
  verifyPassword: (token, password) => api.post(`/public/shares/${token}/verify`, { password }),
  downloadShare: (token, password = null) => {
    return api.post(`/public/shares/${token}/download`, 
      { password }, 
      { responseType: 'blob' }
    );
  },
  getDocuments: (token, password = null) => api.post(`/public/shares/${token}/documents`, { password }),
  downloadSingleDocument: (token, docId, password = null) => {
    return api.post(`/public/shares/${token}/documents/${docId}/download`, 
      { password }, 
      { responseType: 'blob' }
    );
  }
};

export default publicShareService;
